src/uipath/_cli/_evals/_runtime.py

```diff
@@ -601,7 +631,58 @@ async def run_evaluator(
 
         return result
 
-    def _load_evaluators(
+    async def _get_agent_model(self) -> str | None:
+        """Get agent model from the runtime.
+
+        Creates a temporary runtime to query the agent's configured LLM model.
+        The runtime (or one of its delegates) implements LLMAgentRuntimeProtocol
+        which provides the model.
+
+        Returns:
+            The model name from agent settings, or None if not found.
+        """
+        try:
+            temp_runtime = await self.factory.new_runtime(
+                entrypoint=self.context.entrypoint or "",
+                runtime_id="model-query",
+            )
+            try:
+                model = self._find_agent_model_in_runtime(temp_runtime)
+                if model:
+                    logger.debug(f"Got agent model from runtime: {model}")
+                return model
+            finally:
+                await temp_runtime.dispose()
+        except Exception:
+            return None
+
+    def _find_agent_model_in_runtime(self, runtime: Any) -> str | None:
+        """Recursively search for get_agent_model in runtime and its delegates.
+
+        Runtimes may be wrapped (e.g., ResumableRuntime wraps TelemetryWrapper
+        which wraps the base runtime). This method traverses the wrapper chain
+        to find a runtime that implements LLMAgentRuntimeProtocol.
+
+        Args:
+            runtime: The runtime to check (may be a wrapper)
+
+        Returns:
+            The model name if found, None otherwise.
+        """
+        # Check if this runtime implements the protocol
+        if isinstance(runtime, LLMAgentRuntimeProtocol):
+            return runtime.get_agent_model()
+
+        # Check for delegate property (used by UiPathResumableRuntime, TelemetryRuntimeWrapper)
```

Comment:
I'd prefer adding this method in UiPathResumableRuntime and TelemetryRuntimeWrapper but we can do this later..